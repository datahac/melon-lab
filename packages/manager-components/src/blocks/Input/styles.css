.input {
  position: relative;
}

.input__field,
:global(.input__field) {
  font-family: var(--font-family-primary);
  font-size: var(--font-size-m);
  min-width: 100%;
  border: 1px solid var(--color-border);
  padding: var(--space-xs);
  line-height: 1;
  min-height: 38px;
  background-color: var(--color-white);
  -webkit-appearance: none;
  border-radius: 0;
}

.input__field::placeholder,
:global(.input__field)::placeholder {
  color: var(--color-grey);
}

.input__field:disabled,
:global(.input__field):disabled {
  background-color: var(--color-secondary);
}

.input__label {
  display: block;
  margin: 0 0 var(--space-xxs) var(--space-xs);
  font-size: var(--font-size-s);
  line-height: 16px;
}

.input--inside-label :global(.input__field) {
  text-align: right;
}

.input--inside-label .input__label {
  position: absolute;
  top: var(--space-s);
  margin-bottom: 0;
  color: var(--color-primary-dark);
}

.input__field[type='number']::-webkit-outer-spin-button,
.input__field[type='number']::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.input--has-error {
  background-color: var(--color-loss);
}

.input--has-error :global(.input__error) {
  display: block;
}

.input--has-error :global(.input__field) {
  border: 1px solid var(--color-loss);
}

.input__error {
  display: none;
  font-size: var(--font-size-s);
  color: var(--color-white);
  padding: var(--space-xs);
  line-height: 1;
}
