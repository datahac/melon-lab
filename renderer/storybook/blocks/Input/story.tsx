import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react';
import React from 'react';
import Input from './index';

const data = {
  name: 'input',
  type: 'text',
};

storiesOf('Blocks|Input', module)
  .add('Default', () => {
    return <Input {...data} onChange={action('changed')} />;
  })
  .add('Disabled', () => {
    return <Input {...data} disabled={true} />;
  })
  .add('With placeholder', () => {
    return <Input {...data} placeholder="Placeholder..." />;
  })
  .add('With label', () => {
    return <Input {...data} placeholder="Placeholder..." label="Label" />;
  })
  .add('With inside label', () => {
    return (
      <Input {...data} placeholder="0.000" label="Label" insideLabel={true} />
    );
  });
