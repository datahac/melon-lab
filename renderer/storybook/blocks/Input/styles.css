.input {
  position: relative;
}

.input__field,
:global(.input__field) {
  font-family: var(--font-family-primary);
  font-size: var(--font-size-m);
  min-width: 100%;
  border: 1px solid var(--color-border);
  padding: var(--space-xs);
  line-height: 1;
  height: 38px;
  background-color: var(--color-white);
  -webkit-appearance: none;
  border-radius: 0;
}

.input__field::placeholder,
:global(.input__field)::placeholder {
  color: var(--color-grey);
}

.input__field:disabled,
:global(.input__field):disabled {
  background-color: var(--color-secondary);
}

.input__label {
  display: block;
  margin: 0 0 var(--space-xxs) var(--space-xs);
  font-size: var(--font-size-s);
  line-height: 16px;
}

.input--inside-label :global(.input__field) {
  text-align: right;
}

.input--inside-label .input__label {
  position: absolute;
  top: var(--space-s);
  margin-bottom: 0;
  color: var(--color-primary-dark);
}

.input__field[type='number']::-webkit-outer-spin-button,
.input__field[type='number']::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.input--has-error :global(.input__error) {
  display: block;
}

.input--has-error :global(.input__field) {
  border: 1px solid var(--color-loss);
}

.input__error {
  position: absolute;
  left: 4px;
  top: -4px;
  display: none;
  font-size: var(--font-size-s);
  color: var(--color-loss);
  padding: 0 var(--space-xxs);
  background-color: var(--color-white);
  line-height: 1;
  font-weight: var(--font-weight-bold);
}

.input--center-text :global(.input__field) {
  text-align: center;
}
