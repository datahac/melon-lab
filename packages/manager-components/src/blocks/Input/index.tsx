import classNames from 'classnames';
import React, { StatelessComponent } from 'react';
import * as NumberFormat from 'react-number-format';

import styles from './styles.css';

export interface InputProps {
  disabled?: boolean;
  placeholder?: string;
  onChange?: (value, event) => void;
  onBlur?: () => void;
  label?: string;
  type: string;
  insideLabel?: boolean;
  pattern?: string;
  step?: string;
  formatNumber?: boolean;
  decimals?: number;
  value?: string;
  name: string;
  error?: string;
  hidden?: boolean;
}

const Input: StatelessComponent<InputProps> = ({
  disabled,
  placeholder,
  onChange,
  onBlur,
  label,
  type = 'text',
  insideLabel,
  step,
  pattern,
  decimals = 4,
  value,
  formatNumber,
  name,
  error,
  hidden,
}) => {
  const inputClassNames = classNames('input', {
    'input--inside-label': insideLabel,
    'input--has-error': error,
  });

  const handleChange = (e: any): void => {
    const targetValue = e.target.value;
    if (onChange) {
      onChange({ value: targetValue }, e);
    }
  };

  return (
    <div className={inputClassNames}>
      <style jsx>{styles}</style>
      {label && <span className="input__label">{label}</span>}
      {formatNumber ? (
        <NumberFormat
          hidden={hidden}
          name={name}
          pattern={pattern}
          step={step}
          className="input__field"
          disabled={disabled}
          placeholder={placeholder}
          value={value}
          onValueChange={onChange}
          onBlur={onBlur}
          decimalScale={decimals}
          fixedDecimalScale={true}
          isNumericString={true}
        />
      ) : (
        <input
          hidden={hidden}
          name={name}
          pattern={pattern}
          step={step}
          className="input__field"
          type={type}
          disabled={disabled}
          placeholder={placeholder}
          value={value}
          onChange={handleChange}
          onBlur={onBlur}
        />
      )}
      <div className="input__error">{error}</div>
    </div>
  );
};

export default Input;
