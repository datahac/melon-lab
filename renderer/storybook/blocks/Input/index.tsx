import classNames from 'classnames';
import React, { StatelessComponent } from 'react';
import NumberFormat from 'react-number-format';

import styles from './styles.css';

export interface InputProps {
  decimals?: number;
  disabled?: boolean;
  error?: string;
  formatNumber?: boolean;
  hidden?: boolean;
  insideLabel?: boolean;
  label?: string;
  name: string;
  onBlur?: () => void;
  onChange?: (e) => void;
  pattern?: string;
  placeholder?: string;
  step?: string;
  type?: string;
  value?: string;
  maxlength?: number;
  min?: number;
  max?: number;
  centerText?: boolean;
  required?: boolean;
}

const Input: StatelessComponent<InputProps> = ({
  decimals = 0,
  disabled,
  error,
  formatNumber,
  hidden,
  insideLabel,
  label,
  name,
  onBlur,
  onChange,
  pattern,
  placeholder,
  step,
  type = 'text',
  value,
  maxlength,
  min,
  max,
  centerText,
  required,
}) => {
  const inputClassNames = classNames('input', {
    'input--inside-label': insideLabel,
    'input--has-error': error,
    'input--center-text': centerText,
  });

  return (
    <div className={inputClassNames}>
      <style jsx>{styles}</style>
      {label && <span className="input__label">{label}</span>}
      {formatNumber ? (
        <NumberFormat
          hidden={hidden}
          name={name}
          pattern={pattern}
          step={step}
          className="input__field"
          disabled={disabled}
          placeholder={placeholder}
          value={value}
          onChange={onChange}
          onBlur={onBlur}
          decimalScale={decimals}
          fixedDecimalScale={true}
          isNumericString={true}
          type="text"
          min={min}
          max={max}
          required={required}
        />
      ) : (
        <input
          maxLength={maxlength}
          hidden={hidden}
          name={name}
          pattern={pattern}
          step={step}
          className="input__field"
          type={type}
          disabled={disabled}
          placeholder={placeholder}
          value={value}
          onChange={onChange}
          onBlur={onBlur}
          min={min}
          max={max}
          required={required}
        />
      )}
      {error && <div className="input__error">{error}</div>}
    </div>
  );
};

export default Input;
