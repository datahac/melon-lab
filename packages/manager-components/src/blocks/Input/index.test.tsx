import React from 'react';
import Input from './index';

const mockCallback = jest.fn();
const data = {
  name: 'name',
  type: 'text',
  label: 'label',
};

describe('Input', () => {
  const defaultElement = <Input {...data} />;
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(defaultElement);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly with number', () => {
    wrapper.setProps({ formatNumber: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('onChange event', () => {
    wrapper.setProps({ onChange: mockCallback });
    wrapper
      .find('.input__field')
      .simulate('change', { target: { value: 'test' } });
    expect(mockCallback.mock.calls.length).toBe(1);
    expect(mockCallback.mock.calls[0][0].target.value).toBe('test');
  });
});
