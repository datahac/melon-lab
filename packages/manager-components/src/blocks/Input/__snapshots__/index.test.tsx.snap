// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render correctly 1`] = `
<div
  className="jsx-__scopedHash input"
>
  <JSXStyle
    css="__scoped"
    styleId="__scopedHash"
  />
  <span
    className="jsx-__scopedHash input__label"
  >
    label
  </span>
  <input
    className="jsx-__scopedHash input__field"
    name="input"
    onChange={[Function]}
    type="text"
  />
  <div
    className="jsx-__scopedHash input__error"
  />
</div>
`;

exports[`Input should render correctly with number 1`] = `
<div
  className="jsx-__scopedHash input"
>
  <JSXStyle
    css="__scoped"
    styleId="__scopedHash"
  />
  <span
    className="jsx-__scopedHash input__label"
  >
    label
  </span>
  <[object Object]
    allowEmptyFormatting={false}
    allowNegative={true}
    className="input__field"
    decimalScale={4}
    decimalSeparator="."
    displayType="input"
    fixedDecimalScale={true}
    getInputRef={[Function]}
    isAllowed={[Function]}
    isNumericString={true}
    name="input"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseUp={[Function]}
    onValueChange={[Function]}
    prefix=""
    suffix=""
    type="text"
  />
  <div
    className="jsx-__scopedHash input__error"
  />
</div>
`;
